File: docs/demo/V4.x/TESS_PythonAPI_demo/V4公众号案例代码/收费站/MyNet.py
from DLLs.Tessng import PyCustomerNet, tessngIFace, Online, m2p, p2m


class MyNet(PyCustomerNet):
    def __init__(self):
        super(MyNet, self).__init__()
        self.iface = tessngIFace()
        self.netiface = self.iface.netInterface()
        self.simuiface = self.iface.simuInterface()

    def afterLoadNet(self):
        # 创建停车分布
        time_dis_list = [
            self._create_mtc_time_detail(3, 1),
            self._create_mtc_time_detail(5, 1),
        ]
        speed_dis_list: list = [
            self._create_etc_speed_detail(15, 1),
            self._create_etc_speed_detail(20, 1),
        ]
        # 车型ID
        vehicle_type_id: int = 1
        toll_parking_time = self._create_toll_parking_time(vehicle_type_id, time_dis_list, speed_dis_list)
        toll_parking_time_dis = self._create_toll_parking_time_dis([toll_parking_time], "新建停车分布")
        print(f"创建了停车分布，id={toll_parking_time_dis.id}\n")

        # 创建收费车道
        toll_lanes = []
        for lane_number in range(12):
            # 创建收费区域
            toll_point = self._create_toll_point(m2p(40), toll_type=1)
            link_id: int = 1
            location: float = m2p(30)
            length: float = m2p(40)
            toll_lane = self._create_toll_lane(link_id, location, length, lane_number, 0, 3600, [toll_point])
            toll_lanes.append(toll_lane)
            print(f"在序号为{lane_number}的车道上创建了收费车道，id={toll_lane.id()}")
        print()

        # 创建收费路径决策点
        link_id: int = 2
        location: float = m2p(30)
        toll_deci_point = self._create_toll_decision_point(link_id, location)
        print(f"创建了收费路径决策点，id={toll_deci_point.id()}\n")

        # 创建收费路径
        for toll_lane in toll_lanes:
            toll_routing = self._create_toll_routing(toll_deci_point, toll_lane)
            print(f"在收费路径决策点{toll_deci_point.id()}创建了收费路径，id={toll_routing.id()}")
        print()

        # 更新收费决策点的车型对车道选择的分布比重
        toll_routing_id = toll_routing.id()
        toll_lane_id = toll_lane.id()
        toll_deci_point = self.netiface.tollDecisionPoints()[0]
        ect_toll_info = self._create_ect_toll_info(toll_routing_id, toll_lane_id, 0.5)
        vehicle_dis_detail = self._create_vehicle_toll_dis_detail(toll_routing_id, toll_lane_id, 0.5)
        vehicle_dis_info = self._create_vehicle_toll_dis_info(1, [vehicle_dis_detail])
        routing_dis_info = self._create_routing_dis_info([ect_toll_info], [vehicle_dis_info])
        toll_dis_info = self._create_toll_dis_info(None, [routing_dis_info])
        toll_deci_point.updateTollDisInfoList([toll_dis_info])
        print(f"更新了收费决策点的车型对车道选择的分布比重\n")

        # 查看所有收费停车分布
        toll_parking_time_dis = self.netiface.tollParkingTimeDis()
        print(f"收费停车分布的数量是：{len(toll_parking_time_dis)}")
        # 查看所有收费车道
        toll_lanes = self.netiface.tollLanes()
        print(f"收费车道的数量是：{len(toll_lanes)}")
        # 查看所有收费路径决策点
        toll_deci_points = self.netiface.tollDecisionPoints()
        print(f"收费路径决策点的数量是：{len(toll_deci_points)}")

        # 查看收费分布信息列表
        print("收费分布信息列表")
        toll_deci_point = self.netiface.tollDecisionPoints()[0]
        toll_dis_info_list = toll_deci_point.tollDisInfoList()
        for toll_dis_info in toll_dis_info_list:
            # 静态决策路径
            route = toll_dis_info.pIRouting
            print(f"\t静态决策路径：{route}")
            # 收费需求设置
            dis_toll_info_list = toll_dis_info.disTollInfoList
            for dis_toll_info in dis_toll_info_list:
                # 开始时间，单位：秒
                start_time = dis_toll_info.startTime
                # 结束时间，单位：秒
                end_time = dis_toll_info.endTime
                print(f"\t\t开始时间：{start_time}s")
                print(f"\t\t结束时间：{end_time}s")
                # 用户编辑混合车道的ETC车辆比率
                ect_toll_info_list = dis_toll_info.ectTollInfoList
                for ect_toll_info in ect_toll_info_list:
                    # 收费路径编号
                    toll_routing_id = ect_toll_info.tollRoutingID
                    # 收费车道编号
                    toll_lane_id = ect_toll_info.tollLaneID
                    # 车道内ETC占比
                    etc_ratio = ect_toll_info.etcRatio
                    print(f"\t\t\t收费路径ID：{toll_routing_id}")
                    print(f"\t\t\t\t收费车道ID：{toll_lane_id}")
                    print(f"\t\t\t\tETC占比：{etc_ratio}")
                # 车型对车道选择的分布比重
                vehicle_dis_info_list = dis_toll_info.vehicleDisInfoList
                for vehicle_dis_info in vehicle_dis_info_list:
                    # 车型ID
                    vehicle_type_id = vehicle_dis_info.vehicleType
                    print(f"\t\t\t车型ID：{vehicle_type_id}")
                    # 选择分布列表
                    choice_dis_list = vehicle_dis_info.list
                    for choice_dis in choice_dis_list:
                        # 收费路径编号
                        toll_routing_id = choice_dis.tollRoutingID
                        # 收费车道编号
                        toll_lane_id = choice_dis.tollLaneID
                        # 车辆分布比重
                        prop = choice_dis.prop
                        print(f"\t\t\t\t收费路径ID：{toll_routing_id}")
                        print(f"\t\t\t\t收费车道ID：{toll_lane_id}")
                        print(f"\t\t\t\t分布比重：{prop}")
        print()

    def _create_mtc_time_detail(self, time: float, prop: float):
        """创建MTC时间分布"""
        mtc_time_detail = Online.TollStation.DynaMtcTimeDetail()
        # 停车收费时长，单位：m
        mtc_time_detail.time = time
        # 占比
        mtc_time_detail.prop = prop
        return mtc_time_detail

    def _create_etc_speed_detail(self, speed: float, prop: float):
        """创建ETC速度分布"""
        etc_speed_detail = Online.TollStation.DynaEtcSpeedDetail()
        # 通过速度，单位：km/h
        etc_speed_detail.limitSpeed = speed
        # 占比
        etc_speed_detail.prop = prop
        return etc_speed_detail

    def _create_toll_parking_time(self, vehicle_type_id: int, time_dis_list: list, speed_dis_list: list):
        """为一种车型创建MTC停车时间分布和ECT速度分布"""
        toll_parking_time = Online.TollStation.DynaTollParkingTime()
        # 车型ID
        toll_parking_time.vehicleTypeId = vehicle_type_id
        # MTC时间分布
        toll_parking_time.timeDisList = time_dis_list
        # ETC速度分布
        toll_parking_time.speedDisList = speed_dis_list
        return toll_parking_time

    def _create_toll_parking_time_dis(self, parking_time_list: list, name: str = ""):
        """创建收费时间分布"""
        toll_parking_time_dis = Online.TollStation.DynaTollParkingTimeDis()
        # 收费时间分布的名称
        toll_parking_time_dis.name = name
        # 各车型的分布
        toll_parking_time_dis.parkingTimeList = parking_time_list
        time_dis = self.netiface.createTollParkingTimeDis(toll_parking_time_dis)
        return time_dis

    def _create_toll_point(self, distance: float, toll_type: int = 1, time_dis_id: int = 1):
        """创建收费区域"""
        TollStation = Online.TollStation.DynaTollPoint()
        # 距离路段起点的距离，单位：m
        TollStation.location = distance
        # 收费类型：1=MTC，2=ETC，3=ETC&MTC
        TollStation.tollType = toll_type
        # 停车时间分布的ID
        TollStation.timeDisId = time_dis_id
        # 是否启用
        TollStation.enable = True
        return TollStation

    def _create_toll_lane(self, link_id: int, location: float, length: float, lane_number: int, start_time: float, end_time: float, toll_point_list: list, toll_point_length: float = 8):
        """创建收费车道"""
        toll_lane_param = Online.TollStation.DynaTollLane()
        # toll_lane_param.name = "123"
        # 路段ID
        toll_lane_param.roadId = link_id
        # 距离路段起点的距离，单位：m
        toll_lane_param.location = location
        # 收费车道长度，单位：m
        toll_lane_param.length = length
        # 车道序号，从右向左，从0开始
        toll_lane_param.laneNumber = lane_number
        # 开始时间，单位：秒
        toll_lane_param.startTime = start_time
        # 结束时间，单位：秒
        toll_lane_param.endTime = end_time
        # 收费区域列表
        toll_lane_param.tollPoint = toll_point_list
        # 收费区域长度
        toll_lane_param.tollPointLen = toll_point_length
        # 创建收费车道
        toll_lane = self.netiface.createTollLane(toll_lane_param)
        return toll_lane

    def _create_toll_decision_point(self, link_id: int, location: float, name: str = ""):
        """创建收费路径决策点"""
        link = self.netiface.findLink(link_id)
        if not link:
            return None
        toll_deci_point = self.netiface.createTollDecisionPoint(link, location, name)
        return toll_deci_point

    def _create_toll_routing(self, toll_deci_point, toll_lane):
        """创建收费路径"""
        toll_routing = self.netiface.createTollRouting(toll_deci_point, toll_lane)
        return toll_routing

    def _create_ect_toll_info(self, toll_routing_id: int, toll_lane_id: int, etc_ratio: float):
        """创建ETC收费信息"""
        ect_toll_info = Online.TollStation.DynaEtcTollInfo()
        ect_toll_info.tollRoutingID = toll_routing_id
        ect_toll_info.tollLaneID = toll_lane_id
        ect_toll_info.etcRatio = etc_ratio
        return ect_toll_info

    def _create_vehicle_toll_dis_detail(self, toll_routing_id: int, toll_lane_id: int, prop: float):
        """创建某一车型对某一车道选择的分布比重"""
        vehicle_dis_info = Online.TollStation.DynaVehicleTollDisDetail()
        vehicle_dis_info.toll_routing_id = toll_routing_id
        vehicle_dis_info.tollLaneID = toll_lane_id
        vehicle_dis_info.prop = prop
        return vehicle_dis_info

    def _create_vehicle_toll_dis_info(self, vehicle_type_id: int, vehicle_toll_dis_detail_list: list):
        """创建某一车型对所有车道选择的分布比重"""
        vehicle_dis_info = Online.TollStation.DynaVehicleTollDisInfo()
        vehicle_dis_info.vehicleType = vehicle_type_id
        vehicle_dis_info.list = vehicle_toll_dis_detail_list
        return vehicle_dis_info

    def _create_routing_dis_info(self, ect_toll_info_list: list, vehicle_dis_info_list: list):
        """创建路径分布信息"""
        routing_dis_info = Online.TollStation.DynaRoutingDisTollInfo()
        # routing_dis_info.startTime = start_time
        # routing_dis_info.endTime = end_time
        routing_dis_info.ectTollInfoList = ect_toll_info_list
        routing_dis_info.vehicleDisInfoList = vehicle_dis_info_list
        return routing_dis_info

    def _create_toll_dis_info(self, route, routing_dis_info_list: list):
        """创建收费信息"""
        toll_dis_info = Online.TollStation.DynaTollDisInfo()
        toll_dis_info.pIRouting = route
        toll_dis_info.disTollInfoList = routing_dis_info_list
        return toll_dis_info

File: docs/demo/V4.x/TESS_JavaAPI_demo/src/Fundamental_Functions/Toll_Station_Simulation/MyNet.java
//package Fundamental_Functions.Toll_Station_Simulation;
//
//import java.util.ArrayList;
//import java.util.Arrays;
//import java.util.List;
//import com.jidatraffic.tessng.*;
//
//import static com.jidatraffic.tessng.TESSNG.m2p;
//
///**
// * 自定义路网类，实现收费站点相关配置
// */
//public class MyNet extends JCustomerNet {
//    // TESS NG核心接口
//    private TessInterface iface;
//    private NetInterface netiface;
//    private SimuInterface simuiface;
//
//    /**
//     * 构造方法，初始化接口
//     */
//    public MyNet() {
//        super();
//        iface = TESSNG.tessngIFace();
//        netiface = iface.netInterface();
//        simuiface = iface.simuInterface();
//    }
//
//    /**
//     * 路网加载后调用的方法，创建收费相关设施
//     */
//    @Override
//    public void afterLoadNet() {
//        // 创建停车分布
//        List<DynaMtcTimeDetail> timeDisList = new ArrayList<>();
//        timeDisList.add(createMtcTimeDetail(3, 1));
//        timeDisList.add(createMtcTimeDetail(5, 1));
//
//        List<DynaEtcSpeedDetail> speedDisList = new ArrayList<>();
//        speedDisList.add(createEtcSpeedDetail(15, 1));
//        speedDisList.add(createEtcSpeedDetail(20, 1));
//
//        // 车型ID为1的停车时间分布
//        DynaTollParkingTime tollParkingTime = createTollParkingTime(1, timeDisList, speedDisList);
//        List<DynaTollParkingTime> parkingTimeList = new ArrayList<>();
//        parkingTimeList.add(tollParkingTime);
//
//        // 创建收费停车分布
//        DynaTollParkingTimeDis tollParkingTimeDis = createTollParkingTimeDis(parkingTimeList, "新建停车分布");
//        System.out.printf("创建了停车分布，id=%d%n%n", tollParkingTimeDis.getId());
//
//        // 创建收费车道
//        List<ITollLane> tollLanes = new ArrayList<>();
//        for (int laneNumber = 0; laneNumber < 12; laneNumber++) {
//            // 创建收费区域
//            DynaTollPoint tollPoint = createTollPoint(m2p(40), 1);
//
//            int linkId = 1;
//            double location = m2p(30);
//            double length = m2p(40);
//
//            // 使用Arrays.asList()替代List.of()
//            ITollLane tollLane = createTollLane(linkId, location, length, laneNumber, 0, 3600,
//                    new ArrayList<>(Arrays.asList(tollPoint)));
//            tollLanes.add(tollLane);
//            System.out.printf("在序号为%d的车道上创建了收费车道，id=%d%n", laneNumber, tollLane.id());
//        }
//        System.out.println();
//
//        // 创建收费路径决策点
//        int linkId = 2;
//        double location = m2p(30);
//        ITollDecisionPoint tollDeciPoint = createTollDecisionPoint(linkId, location);
//        System.out.printf("创建了收费路径决策点，id=%d%n%n", tollDeciPoint.id());
//
//        // 创建收费路径
//        for (ITollLane tollLane : tollLanes) {
//            ITollRouting tollRouting = createTollRouting(tollDeciPoint, tollLane);
//            System.out.printf("在收费路径决策点%d创建了收费路径，id=%d%n", tollDeciPoint.id(), tollRouting.id());
//        }
//        System.out.println();
//
//        // 更新收费决策点的车型对车道选择的分布比重
//        if (!tollLanes.isEmpty()) {
//            ITollLane lastTollLane = tollLanes.get(tollLanes.size() - 1);
//            List<ITollRouting> tollRoutings = netiface.tol();
//            if (!tollRoutings.isEmpty()) {
//                ITollRouting lastTollRouting = tollRoutings.get(tollRoutings.size() - 1);
//
//                List<ITollDecisionPoint> tollDeciPoints = netiface.tollDecisionPoints();
//                if (!tollDeciPoints.isEmpty()) {
//                    ITollDecisionPoint firstTollDeciPoint = tollDeciPoints.get(0);
//
//                    DynaEtcTollInfo ectTollInfo = createEctTollInfo((int)lastTollRouting.id(), (int)lastTollLane.id(), 0.5);
//                    DynaVehicleTollDisDetail vehicleDisDetail = createVehicleTollDisDetail((int)lastTollRouting.id(), (int)lastTollLane.id(), 0.5);
//
//                    // 使用Arrays.asList()替代List.of()
//                    DynaVehicleTollDisInfo vehicleDisInfo = createVehicleTollDisInfo(1, new ArrayList<>(Arrays.asList(vehicleDisDetail)));
//                    DynaRoutingDisTollInfo routingDisInfo = createRoutingDisInfo(new ArrayList<>(Arrays.asList(ectTollInfo)),
//                            new ArrayList<>(Arrays.asList(vehicleDisInfo)));
//                    DynaTollDisInfo tollDisInfo = createTollDisInfo(null, new ArrayList<>(Arrays.asList(routingDisInfo)));
//
//                    firstTollDeciPoint.updateTollDisInfoList(new ArrayList<>(Arrays.asList(tollDisInfo)));
//                    System.out.println("更新了收费决策点的车型对车道选择的分布比重\n");
//                }
//            }
//        }
//
//        // 查看所有收费相关设施数量
//        List<IDynaTollParkingTimeDis> allTollParkingTimeDis = netiface.tollParkingTimeDis();
//        System.out.printf("收费停车分布的数量是：%d%n", allTollParkingTimeDis.size());
//
//        List<ITollLane> allTollLanes = netiface.tollLanes();
//        System.out.printf("收费车道的数量是：%d%n", allTollLanes.size());
//
//        List<ITollDecisionPoint> allTollDeciPoints = netiface.tollDecisionPoints();
//        System.out.printf("收费路径决策点的数量是：%d%n", allTollDeciPoints.size());
//
//        // 查看收费分布信息列表
//        System.out.println("收费分布信息列表");
//        if (!allTollDeciPoints.isEmpty()) {
//            ITollDecisionPoint firstTollDeciPoint = allTollDeciPoints.get(0);
//            List<DynaTollDisInfo> tollDisInfoList = firstTollDeciPoint.tollDisInfoList();
//
//            for (DynaTollDisInfo tollDisInfo : tollDisInfoList) {
//                // 静态决策路径
//                IRouting route = tollDisInfo.pIRouting();
//                System.out.printf("\t静态决策路径：%s%n", route);
//
//                // 收费需求设置
//                List<DynaRoutingDisTollInfo> disTollInfoList = tollDisInfo.disTollInfoList();
//                for (DynaRoutingDisTollInfo disTollInfo : disTollInfoList) {
//                    // 开始时间和结束时间（单位：秒）
//                    System.out.printf("\t\t开始时间：%ds%n", disTollInfo.startTime());
//                    System.out.printf("\t\t结束时间：%ds%n", disTollInfo.endTime());
//
//                    // ETC车辆比率
//                    List<DynaEtcTollInfo> ectTollInfoList = disTollInfo.ectTollInfoList();
//                    for (DynaEtcTollInfo ectTollInfo : ectTollInfoList) {
//                        System.out.printf("\t\t\t收费路径ID：%d%n", ectTollInfo.tollRoutingID());
//                        System.out.printf("\t\t\t\t收费车道ID：%d%n", ectTollInfo.tollLaneID());
//                        System.out.printf("\t\t\t\tETC占比：%f%n", ectTollInfo.etcRatio());
//                    }
//
//                    // 车型对车道选择的分布比重
//                    List<DynaVehicleTollDisInfo> vehicleDisInfoList = disTollInfo.vehicleDisInfoList();
//                    for (DynaVehicleTollDisInfo vehicleDisInfo : vehicleDisInfoList) {
//                        System.out.printf("\t\t\t车型ID：%d%n", vehicleDisInfo.vehicleType());
//
//                        List<DynaVehicleTollDisDetail> choiceDisList = vehicleDisInfo.list();
//                        for (DynaVehicleTollDisDetail choiceDis : choiceDisList) {
//                            System.out.printf("\t\t\t\t收费路径ID：%d%n", choiceDis.tollRoutingID());
//                            System.out.printf("\t\t\t\t收费车道ID：%d%n", choiceDis.tollLaneID());
//                            System.out.printf("\t\t\t\t分布比重：%f%n", choiceDis.prop());
//                        }
//                    }
//                }
//            }
//        }
//        System.out.println();
//    }
//
//    /**
//     * 创建MTC时间分布
//     */
//    private DynaMtcTimeDetail createMtcTimeDetail(float time, float prop) {
//        DynaMtcTimeDetail mtcTimeDetail = new DynaMtcTimeDetail();
//        mtcTimeDetail.setTime(time);  // 停车收费时长（单位：分钟）
//        mtcTimeDetail.setProp(prop);  // 占比
//        return mtcTimeDetail;
//    }
//
//    /**
//     * 创建ETC速度分布
//     */
//    private DynaEtcSpeedDetail createEtcSpeedDetail(float speed, float prop) {
//        DynaEtcSpeedDetail etcSpeedDetail = new DynaEtcSpeedDetail();
//        etcSpeedDetail.setLimitSpeed(speed);  // 通过速度（单位：km/h）
//        etcSpeedDetail.setProp(prop);  // 占比
//        return etcSpeedDetail;
//    }
//
//    /**
//     * 为一种车型创建MTC停车时间分布和ETC速度分布
//     */
//    private DynaTollParkingTime createTollParkingTime(int vehicleTypeId,
//                                                      List<DynaMtcTimeDetail> timeDisList,
//                                                      List<DynaEtcSpeedDetail> speedDisList) {
//        DynaTollParkingTime tollParkingTime = new DynaTollParkingTime();
//        tollParkingTime.setVehicleTypeId(vehicleTypeId);  // 车型ID
//        tollParkingTime.setTimeDisList(timeDisList);      // MTC时间分布
//        tollParkingTime.setSpeedDisList(speedDisList);    // ETC速度分布
//        return tollParkingTime;
//    }
//
//    /**
//     * 创建收费时间分布
//     */
//    private DynaTollParkingTimeDis createTollParkingTimeDis(List<DynaTollParkingTime> parkingTimeList, String name) {
//        DynaTollParkingTimeDis tollParkingTimeDis = new DynaTollParkingTimeDis();
//        tollParkingTimeDis.setName(name);                // 收费时间分布名称
//        tollParkingTimeDis.setParkingTimeList(parkingTimeList);  // 各车型的分布
//        return netiface.createTollParkingTimeDis(tollParkingTimeDis);
//    }
//
//    /**
//     * 创建收费区域
//     */
//    private DynaTollPoint createTollPoint(double distance, int tollType) {
//        return createTollPoint(distance, tollType, 1);
//    }
//
//    private DynaTollPoint createTollPoint(double distance, int tollType, int timeDisId) {
//        DynaTollPoint tollPoint = new DynaTollPoint();
//        tollPoint.setLocation(distance);  // 距离路段起点的距离（单位：米）
//        tollPoint.setTollType(tollType);  // 收费类型：1=MTC，2=ETC，3=ETC&MTC
//        tollPoint.setTimeDisId(timeDisId);  // 停车时间分布的ID
//        tollPoint.setEnable(true);        // 启用收费区域
//        return tollPoint;
//    }
//
//    /**
//     * 创建收费车道
//     */
//    private ITollLane createTollLane(int linkId, double location, double length, int laneNumber,
//                                     float startTime, float endTime, List<DynaTollPoint> tollPointList) {
//        return createTollLane(linkId, location, length, laneNumber, startTime, endTime, tollPointList, 8);
//    }
//
//    private ITollLane createTollLane(int linkId, double location, double length, int laneNumber,
//                                     float startTime, float endTime, List<DynaTollPoint> tollPointList,
//                                     float tollPointLength) {
//        DynaTollLane tollLaneParam = new DynaTollLane();
//        tollLaneParam.setRoadId(linkId);         // 路段ID
//        tollLaneParam.setLocation(location);     // 距离路段起点的距离（单位：米）
//        tollLaneParam.setLength(length);         // 收费车道长度（单位：米）
//        tollLaneParam.setLaneNumber(laneNumber); // 车道序号（从右向左，从0开始）
//        tollLaneParam.setStartTime(startTime);   // 开始时间（单位：秒）
//        tollLaneParam.setEndTime(endTime);       // 结束时间（单位：秒）
//        tollLaneParam.setTollPoint(tollPointList); // 收费区域列表
//        tollLaneParam.setTollPointLen(tollPointLength); // 收费区域长度
//        return netiface.createTollLane(tollLaneParam);
//    }
//
//    /**
//     * 创建收费路径决策点
//     */
//    private ITollDecisionPoint createTollDecisionPoint(int linkId, double location) {
//        return createTollDecisionPoint(linkId, location, "");
//    }
//
//    private ITollDecisionPoint createTollDecisionPoint(int linkId, double location, String name) {
//        ILink link = netiface.findLink(linkId);
//        if (link == null) return null;
//        return netiface.createTollDecisionPoint(link, location, name);
//    }
//
//    /**
//     * 创建收费路径
//     */
//    private ITollRouting createTollRouting(ITollDecisionPoint tollDeciPoint, ITollLane tollLane) {
//        return netiface.createTollRouting(tollDeciPoint, tollLane);
//    }
//
//    /**
//     * 创建ETC收费信息
//     */
//    private DynaEtcTollInfo createEctTollInfo(int tollRoutingId, int tollLaneId, double etcRatio) {
//        DynaEtcTollInfo ectTollInfo = new DynaEtcTollInfo();
//        ectTollInfo.setTollRoutingID(tollRoutingId);  // 收费路径编号
//        ectTollInfo.setTollLaneID(tollLaneId);        // 收费车道编号
//        ectTollInfo.setEtcRatio(etcRatio);            // 车道内ETC占比
//        return ectTollInfo;
//    }
//
//    /**
//     * 创建某一车型对某一车道选择的分布比重
//     */
//    private DynaVehicleTollDisDetail createVehicleTollDisDetail(int tollRoutingId, int tollLaneId, double prop) {
//        DynaVehicleTollDisDetail vehicleDisDetail = new DynaVehicleTollDisDetail();
//        vehicleDisDetail.setTollRoutingID(tollRoutingId);  // 收费路径编号
//        vehicleDisDetail.setTollLaneID(tollLaneId);        // 收费车道编号
//        vehicleDisDetail.setProp(prop);                    // 分布比重
//        return vehicleDisDetail;
//    }
//
//    /**
//     * 创建某一车型对所有车道选择的分布比重
//     */
//    private DynaVehicleTollDisInfo createVehicleTollDisInfo(int vehicleTypeId,
//                                                            List<DynaVehicleTollDisDetail> vehicleTollDisDetailList) {
//        DynaVehicleTollDisInfo vehicleDisInfo = new DynaVehicleTollDisInfo();
//        vehicleDisInfo.setVehicleType(vehicleTypeId);  // 车型ID
//        vehicleDisInfo.setList(vehicleTollDisDetailList);  // 选择分布列表
//        return vehicleDisInfo;
//    }
//
//    /**
//     * 创建路径分布信息
//     */
//    private DynaRoutingDisTollInfo createRoutingDisInfo(List<DynaEtcTollInfo> ectTollInfoList,
//                                                        List<DynaVehicleTollDisInfo> vehicleDisInfoList) {
//        DynaRoutingDisTollInfo routingDisInfo = new DynaRoutingDisTollInfo();
//        routingDisInfo.setEctTollInfoList(ectTollInfoList);      // ETC收费信息列表
//        routingDisInfo.setVehicleDisInfoList(vehicleDisInfoList);  // 车型分布信息列表
//        return routingDisInfo;
//    }
//
//    /**
//     * 创建收费信息
//     */
//    private DynaTollDisInfo createTollDisInfo(IRouting route, List<DynaRoutingDisTollInfo> routingDisInfoList) {
//        DynaTollDisInfo tollDisInfo = new DynaTollDisInfo();
//        tollDisInfo.setPIRouting(route);                // 静态决策路径
//        tollDisInfo.setDisTollInfoList(routingDisInfoList);  // 路径分布信息列表
//        return tollDisInfo;
//    }
//
//
//}